jekyll/_posts/2020-10-16-version-140-released.md

```diff
@@ -0,0 +1,1076 @@
+
+
+### Known incompatibilities and gotchas
+
+#### Deepcopy
+
+If you use `system.deepCopy` in your code, you need to enable it via
+`--deepCopy:on` on the command line.
+This is a band-aid and a better solution for this will arrive in the future.
+
+The reason for this opt-in switch is that ARC and ORC do not use the old runtime
+type information (RTTI) that `system.deepCopy` is built upon.
+We've built ARC and ORC with embedded devices in mind where the overhead of the
+old RTTI has been reported to be unacceptable.
+
+
+#### Order of declarations
+
+The second gotcha people might encounter is the strict ordering requirement
+for custom destructors; this affects old code as well, because finalizers are
+mapped to destructors.
+For example:
+
+```nim
+type
+  CustomObject = ref object
+    p: ptr int
+
+proc tricky() =
+  # would use the default destructor
+  var x: CustomObject
+
+proc finalizer(x: CustomObject) =
+  if x.p != nil:
+    dealloc(x.p)
+
+proc newCustomObject(): CustomObject =
+  new(result, finalizer)
+```
+
+This produces the following error message:
+```
+temp.nim(15, 6) Error: cannot bind another '=destroy' to: CustomObject:ObjectType;
+previous declaration was constructed here implicitly: temp.nim(8, 7)
+```
+
+The reason is that only after the `new(result, finalizer)` call the compiler knows
+that it needs to generate a custom destructor for the `CustomObject` type,
+but when it compiled `tricky` the default destructor was used.
+
+The solution is to move the proc `tricky` after the `newCustomObject` declaration:
+```nim
+type
+  CustomObject = ref object
+    p: ptr int
+
+proc finalizer(x: CustomObject) =
+  if x.p != nil:
+    dealloc(x.p)
+
+proc newCustomObject(): CustomObject =
+  new(result, finalizer)
+
+proc tricky() =
+  # uses the custom destructor
+  var x: CustomObject
+```
+
+This is a current limitation that will be improved in future versions of the compiler.
+
+
+
+## Strict(er) functions
+
+This is an experimental feature available via `{.experimental: "strictFuncs".}` pragma in
+your code or via the `--experimental:strictFuncs` switch.
+
+It introduces a stricter definition of a "side effect":
+If an object is reachable via a parameter that is not declared as a `var` parameter,
+any mutation to that object counts as a side effect.
+
+Here is an example from
+[the experimental manual](https://nim-lang.github.io/Nim/manual_experimental.html#strict-funcs):
+
+```nim
+{.experimental: "strictFuncs".}
+
+type
+  Node = ref object
+    le, ri: Node
+    data: string
+
+func len(n: Node): int =
+  # valid: len does not have side effects
+  var it = n
+  while it != nil:
+    inc result
+    it = it.ri
+
+func mut(n: Node) =
+  let m = n # is the statement that connected the mutation to the parameter
+  m.data = "yeah" # the mutation is here
+  # Error: 'mut' can have side effects
+  # an object reachable from 'n' is potentially mutated
+```
+
+
```

Comment:
```suggestion
For more details, please see [Araq's blog post](https://nim-lang.org/blog/2020/09/01/write-tracking.html).

```
Or something similar.

The experimental manual doesn't really give much background information, and we linked to @Yardanico's post about ARC/ORC so I think it makes sense to mention the "write tracking (part 2)" post here.